### DISCUSSION TITLE
Any way to set mode from a phony target?

### DISCUSSION THREAD

**Discussion by dkaszews** [xmake-io/xmake-repo#5721] [Ideas]
I have the following target for running coverage (and similar for performance tests):

```lua
target('coverage')
    set_kind('phony')
    on_run(function (target)
        assert(is_mode('coverage'), 'Incorrect mode, please set to coverage')
        os.exec('xmake build')
        os.exec('xmake test -v')
        os.exec('gcovr')
    end)
```

Is there a way for the target to instead set the mode to correct one? I tried `os.exec('xmake f -m coverage')`, but it does not seem to affect the following build and test until it is run again.

**Comment by waruqi** [comment]
you should use task to do it.

**Reply by dkaszews** [reply]
Could you please elaborate? Should I replace the target with a task, or create a separate task for setting mode? How that does that one differ from just running a command from target `on_run`?

I switched `target('coverage')` to `task('coverage')`, but `xmake coverage` results in:
> error: invalid task: coverage

**Reply by waruqi** [reply]
you need add set_menu

https://xmake.io/#/manual/plugin_task

**Reply by dkaszews** [reply]
Thank you!

For any who find it, I am now using the following:

```lua
task('coverage')
    on_run(function (target)
        os.exec('xmake config --mode=coverage')
        os.exec('xmake build -v')
        os.exec('xmake test -v')
        os.exec('gcovr')
    end)
    set_menu {}
```

And running it with `xmake coverage`.
